lib/App/Sqitch/Command/check.pm

```diff
@@ -0,0 +1,208 @@
+    is  => 'ro',
+    isa => Str,
+);
+
+has target => (
+    is      => 'rw',
+    isa     => Target,
+    handles => [qw(engine plan plan_file)],
+);
+
+has date_format => (
+    is      => 'ro',
+    lazy    => 1,
+    isa     => Str,
+    default => sub {
+        shift->sqitch->config->get( key => 'status.date_format' ) || 'iso'
+    }
+);
+
+has project => (
+    is      => 'ro',
+    isa     => Str,
+    lazy    => 1,
+    default => sub {
+        my $self = shift;
+        try { $self->plan->project } catch {
+            # Just die on parse and I/O errors.
+            die $_ if try { $_->ident eq 'parse' || $_->ident eq 'io' };
+
+            # Try to extract a project name from the registry.
+            my $engine = $self->engine;
+            hurl status => __ 'Database not initialized for Sqitch'
+                unless $engine->initialized;
+            my @projs = $engine->registered_projects
+                or hurl status => __ 'No projects registered';
+            hurl status => __x(
+                'Use --project to select which project to query: {projects}',
+                projects => join __ ', ', @projs,
+            ) if @projs > 1;
+            return $projs[0];
+        };
+    },
+);
+
+sub options {
+    return qw(
+        project=s
+        target|t=s
+    );
+}
+
+sub execute {
+    my $self = shift;
+    my ($targets) = $self->parse_args(
+        target => $self->target_name,
+        args   => \@_,
+    );
+
+    # Warn on multiple targets.
+    my $target = shift @{ $targets };
+    $self->warn(__x(
+        'Too many targets specified; connecting to {target}',
+        target => $target->name,
+    )) if @{ $targets };
+
+    # Good to go.
+    $self->target($target);
+    my $engine = $target->engine;
+
+    # Where are we?
+    $self->comment( __x 'On database {db}', db => $engine->destination );
+
+    # Exit with status 1 on no state, probably not expected.
+    my $state = try {
+        $engine->current_state( $self->project )
+    } catch {
+        # Just die on parse and I/O errors.
+        die $_ if try { $_->ident eq 'parse' || $_->ident eq 'io' };
+
+        # Hrm. Maybe not initialized?
+        die $_ if $engine->initialized;
+        hurl status => __x(
+            'Database {db} has not been initialized for Sqitch',
+            db => $engine->registry_destination
+        );
+    };
+
+    my @deployed_changes = $engine->deployed_changes;
+
+    my %deployed_script_hashes;
+    foreach my $change ($engine->deployed_changes) {
+        $deployed_script_hashes{$change->{'id'}} = $change->{'script_hash'};
+    }
+
+    my $workdir_plan = $target->plan;
+    my @workdir_changes = $workdir_plan->changes;
+    foreach my $change (@workdir_changes) {
+        $self->comment(__x(
+            'Working directory script {script_file} is different from deployed script',
+            script_file => $change->deploy_file
+        )) if $change->script_hash ne $deployed_script_hashes{$change->id};
```

Comment:
We should give some thought to the output of failed checks. Should they be parsable? Have some kind of header or something? Also, once you find one bad ID, all the remaining IDs will be bad, so might as well say something like `"Script signatures diverge at change {change}"` and stop processing. In fact, you should be able to just iterate over the plan and the deployed changes without using a lookup hash, because the first time something is out of order it's all over.

Once you find something that's not right, set up a variable to exit with a non-zero status, as well. I think you can just set a boolean, and then raise an error using `hurl`; maybe have the exit value be the number of failures (just going to be one for now, but as more checks are added, it can go up), something like:

``` perl
        hurl {
            ident => 'check',
            message => __np(
                'Failed one check',
                'Failed {count} checks',
                $num_failed,
                count => $num_failed,
            ),
            exitval => $num_failed,
        } if $num_failed;
```